controllers/kubeadmconfig_controller.go

```diff
@@ -378,15 +377,23 @@ func (r *KubeadmConfigReconciler) ClusterToKubeadmConfigs(o handler.MapObject) [
 		return nil
 	}
 
-	labels := map[string]string{clusterv1.MachineClusterLabelName: c.Name}
+	selectors := []client.ListOption{
+		client.InNamespace(c.Namespace),
+		client.MatchingLabels{
+			clusterv1.MachineClusterLabelName: c.Name,
+		},
+	}
+
 	machineList := &clusterv1.MachineList{}
-	if err := r.List(context.Background(), machineList, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
+	if err := r.List(context.Background(), machineList, selectors...); err != nil {
 		r.Log.Error(err, "failed to list Machines", "Cluster", c.Name, "Namespace", c.Namespace)
 		return nil
 	}
+
 	for _, m := range machineList.Items {
-		if m.Spec.InfrastructureRef.GroupVersionKind() == bootstrapv1.GroupVersion.WithKind("KubeadmConfig") {
-			name := client.ObjectKey{Namespace: m.Namespace, Name: m.Spec.InfrastructureRef.Name}
+		if m.Spec.Bootstrap.ConfigRef != nil &&
+			m.Spec.Bootstrap.ConfigRef.GroupVersionKind() == bootstrapv1.GroupVersion.WithKind("KubeadmConfig") {
+			name := client.ObjectKey{Namespace: m.Namespace, Name: m.Spec.Bootstrap.ConfigRef.Name}
```

Comment:
This diff is hard to read with all the additional refactoring as well as the bugfix, but afaict, this is the heart of the fix here where we were previously mistakenly queuing the InfrastructureRef.Name rather than the ConfigRef.Name. Is that correct?

Reply:
yeah